contrib/golang/filters/http/source/go/pkg/http/filtermanager.go

```diff
@@ -24,18 +24,15 @@ import (
 	"github.com/envoyproxy/envoy/contrib/golang/filters/http/source/go/pkg/api"
 )
 
-var httpFilterConfigFactory = sync.Map{}
+var httpFilterConfigFactoryAndParser = sync.Map{}
 
-func RegisterHttpFilterConfigFactory(name string, f api.StreamFilterConfigFactory) {
-	httpFilterConfigFactory.Store(name, f)
+type filterConfigFactoryAndParser struct {
+	configFactory api.StreamFilterConfigFactory
+	configParser  api.StreamFilterConfigParser
 }
 
-// no parser by default
-var httpFilterConfigParser api.StreamFilterConfigParser = nil
-
-// TODO merge it to api.HttpFilterConfigFactory
-func RegisterHttpFilterConfigParser(parser api.StreamFilterConfigParser) {
-	httpFilterConfigParser = parser
+func RegisterHttpFilterConfigFactoryAndParser(name string, f api.StreamFilterConfigFactory, parser api.StreamFilterConfigParser) {
```

Comment:
It would be helpful to add some comments or code-level checks, such as indicating that the "factory" parameter is required and the "parser" parameter is optional.